src/identityApiClient.ts

```diff
@@ -0,0 +1,284 @@
+        callback: IdentityCallback,
+        originalIdentityApiData: IdentityApiData,
+        parseIdentityResponse: IIdentity['parseIdentityResponse'],
+        mpid: MPID,
+        knownIdentities: UserIdentities
+    ) {
+        const { verbose, error } = mpInstance.Logger;
+        const { invokeCallback } = mpInstance._Helpers;
+
+        verbose(Messages.InformationMessages.SendIdentityBegin);
+        if (!identityApiRequest) {
+            error(Messages.ErrorMessages.APIRequestEmpty);
+            return;
+        }
+        verbose(Messages.InformationMessages.SendIdentityHttp);
+
+        if (mpInstance._Store.identityCallInFlight) {
+            invokeCallback(
+                callback,
+                HTTPCodes.activeIdentityRequest,
+                'There is currently an Identity request processing. Please wait for this to return before requesting again'
+            );
+            return;
+        }
+
+        const previousMPID = mpid || null;
+        const uploadUrl = this.getUploadUrl(method, mpid);
+
+        const uploader: AsyncUploader = window.fetch
+            ? new FetchUploader(uploadUrl)
+            : new XHRUploader(uploadUrl);
+
+        // https://go.mparticle.com/work/SQDSDKS-6568
+        const fetchPayload: IdentityApiRequestPayload = {
+            method: 'post',
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'application/json',
+                'x-mp-key': mpInstance._Store.devToken,
+            },
+            body: JSON.stringify(identityApiRequest),
+        };
+
+        try {
+            mpInstance._Store.identityCallInFlight = true;
+            const response: Response = await uploader.upload(fetchPayload);
+
+            let identityResponse: IIdentityResponse;
+
+            if (response.json) {
+                // https://go.mparticle.com/work/SQDSDKS-6568
+                // FetchUploader returns the response as a JSON object that we have to await
+                const responseBody: IdentityResultBody = await response.json();
+
+                identityResponse = this.getIdentityResponseFromFetch(
+                    response,
+                    responseBody
+                );
+            } else {
+                identityResponse = this.getIdentityResponseFromXHR(
+                    (response as unknown) as XMLHttpRequest
+                );
+            }
+
+            verbose(
+                'Received Identity Response from server: ' +
+                    JSON.stringify(identityResponse.responseText)
+            );
+
+            parseIdentityResponse(
+                identityResponse,
+                previousMPID,
+                callback,
+                originalIdentityApiData,
+                method,
+                knownIdentities,
+                false
+            );
+        } catch (err) {
+            const errorMessage = (err as Error).message || 'Unknown Error';
```

Comment:
I think `Unknown error` is too generic. While we will know it is only coming from here if we do a global search for it, it might be more helpful to provide more context.  

I actually think here is a good instance to do something like a `toString` which you had before if there is no message. That will allow us to debug further.  What do you think?

Reply:
I think I would prefer `err.toString()` if `error.message` is not available.